src/rocksdb/src/lib.rs

```diff
@@ -103,6 +158,20 @@ impl InstanceOptions {
         // Set the env first so tuning applies to the shared `Env`.
         options.set_env(&self.env);
 
+        if let Some((fn_name, merge_fn)) = &self.merge_operator {
+            let bincode = self.bincode.clone();
+            let merge_fn = merge_fn.clone();
+            options.set_merge_operator_associative(fn_name, move |key, existing, operands| {
```

Comment:
does this not set only the full merge fn? You need https://docs.rs/rocksdb/latest/rocksdb/struct.Options.html#method.set_merge_operator to also have a partial one, right?

Reply:
No the point of the associative merge operator is that you only provide one function and it's used for both full/partial merges since the values you're dealing with are always the same. If the values we passed to `merge` were different than `put`, then you need to implement the full merge operator with separate handlers for each.

See https://github.com/facebook/rocksdb/wiki/Merge-Operator#associativity-vs-non-associativity
and specifically:
> AssociativeMergeOperator simply inherits from MergeOperator and provides private default implementations of these methods, while exposing a wrapper function for simplicity.

Reply:
I mention that this should be an ` A merge operator to use for associative merges` above but will leave a comment here to link to the wiki and that quote.

Reply:
thanks!